src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,94 @@
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private Node<K,V>[] baskets;
+    private int currentCapacity;
+
+    public MyHashMap() {
+        currentCapacity = INITIAL_CAPACITY;
+        baskets = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= LOAD_FACTOR * currentCapacity) {
+            resizeBasket();
+        }
+        Node<K,V> inputNode = new Node<>(key, value, null);
+        int basketIndex = getBasketIndex(key);
+        Node<K,V> nodeInBasket = baskets[basketIndex];
+
+        if (nodeInBasket == null) {
+            baskets[basketIndex] = inputNode;
+            size++;
+        } else {
+            while (nodeInBasket.next != null) {
+                if (nodeInBasket.key == key || (key != null && key.equals(nodeInBasket.key))) {
+                    nodeInBasket.value = value;
+                    return;
+                }
+                nodeInBasket = nodeInBasket.next;
+            }
+            if (nodeInBasket.key == key || (key != null && key.equals(nodeInBasket.key))) {
+                nodeInBasket.value = value;
+            } else {
+                nodeInBasket.next = inputNode;
+                size++;
+            }
+        }
 
     }
 
     @Override
     public V getValue(K key) {
+        Node<K,V> node = baskets[getBasketIndex(key)];
+        while (node != null) {
+            if ((node.key == null || key == null)
+                    ? node.key == key : node.key.equals(key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resizeBasket() {
```

Comment:
```suggestion
    private void resize() {
```
that may sound like you wanna resize one basket, which doesn't make sense

Reply:
agree, fixed